File: dynabuffers-python/dynabuffers/ast/structural/ClassOptions.py
class ClassOptionsOptions:
    def __init__(self, primary:bool, deprecated:bool, implicit: bool):
        self.primary = primary
        self.deprecated = deprecated
        self.implicit = implicit


class ClassOptions:

    def __init__(self, options: ClassOptionsOptions):
        self.options = options

    def is_primary(self):
        return self.options.primary

    def is_deprecated(self):
        return self.options.deprecated

    def is_implicit(self):
        return self.options.implicit

File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/structural/ClassOptions.kt
package dynabuffers.ast.structural

import dynabuffers.api.IType

class ClassOptions(private val options: ClassOptionsOptions) : IType {

    data class ClassOptionsOptions(val primary:Boolean, val deprecated: Boolean, val implicit: Boolean)

    fun isPrimary() = options.primary
    fun isDeprecated() = options.deprecated
    fun isImplicit() = options.implicit

}
